envoy_pkg/packages/getenvoy_package.bzl

```diff
@@ -46,153 +46,219 @@ def getenvoy_package(name, binary_target):
         tags = ["manual"],
     )
 
-    pkg_tar(
-        name = "tar-package-stripped",
-        extension = "tar.xz",
-        srcs = [":envoy"],
-        remap_paths = {
-            "/envoy": "bin/envoy",
-        },
-        modes = {
-            "bin/envoy": "0755",
-        },
-        package_dir = _tar_dir(PACKAGE_VERSION),
-        tags = ["manual"],
-    )
+    if PACKAGE_VERSION["variant"] != "istio-proxy-compat":
```

Comment:
can we just add `-istio-compat.tar.gz` to each variant? (i.e. without specializing `istio-proxy-compat` and add new packages), so the envoy package can be used in istio after istio moves to unmodified Envoy.

Reply:
I did this specialization mainly so you couldn't build an "istio-proxy-compat" debian package since istio only consumes tar's (e.g. so someone can't manually try building a debian in istio-proxy-compat, and expect it to work). But I'd be happy to rework if you think it's not worth doing that.

Reply:
Right I meant adding `istio-compat` as a side of `tar`, `deb`, `rpm`, `distroless`. so when you build istio-proxy (with corresponding envs) you can get deb and istio-compat

Reply:
Ah I see what you mean. So not making it a full variant, but just making another arg like: `--build_deb` that's called: `--build_istio_compat`.

Yeah I can rework to do that.

Reply:
SGTM